app/dawn-hub/Views/OnboardingViews/Step2_BlockSettingsView/BlockSettingsView.jsx

```diff
@@ -39,57 +44,56 @@ class BlockSettingsView extends Component {
 
 	componentDidMount() {
 		const { setupLifecycle: { blockSetupSeen } } = this.props;
+
 		if (blockSetupSeen) {
 			const {
-				antiSuite: {
-					enable_anti_tracking,
-					enable_ad_block,
-					enable_smart_block,
-				},
+				enable_anti_tracking,
+				enable_ad_block,
+				enable_smart_block,
 				blockingPolicy
 			} = this.props;
 
-			const decodedPolicy = this.decodeBlockingPolicy(blockingPolicy);
-
 			this.setState({
-				kindsOfTrackers: decodedPolicy,
+				kindsOfTrackers: blockingPolicy,
 				enable_ad_block,
 				enable_anti_tracking,
 				enable_smart_block
 			});
 		}
 	}
 
-	decodeBlockingPolicy = (blockingPolicy) => {
+	componentDidUpdate() {
+		const {
+			recommendedChoices,
+			enable_anti_tracking,
+			enable_ad_block,
+			enable_smart_block,
+			kindsOfTrackers
+		} = this.state;
+
+		if (!recommendedChoices && kindsOfTrackers === BLOCKING_POLICY_RECOMMENDED && enable_ad_block === true && enable_anti_tracking === true && enable_smart_block === true) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({ recommendedChoices: true });
+		} else if (recommendedChoices && (kindsOfTrackers !== BLOCKING_POLICY_RECOMMENDED || enable_ad_block !== true || enable_anti_tracking !== true || enable_smart_block !== true)) {
+			// eslint-disable-next-line react/no-did-update-set-state
+			this.setState({ recommendedChoices: false });
+		}
+	}
+
+	enumerateBlockingPolicy = (blockingPolicy) => {
 		let decodedPolicy;
-		if (typeof blockingPolicy === 'number') {
-			switch (blockingPolicy) {
-				case 1:
-					decodedPolicy = 'BLOCKING_POLICY_EVERYTHING';
-					break;
-				case 2:
-					decodedPolicy = 'BLOCKING_POLICY_RECOMMENDED';
-					break;
-				case 3:
-					decodedPolicy = 'BLOCKING_POLICY_NOTHING';
-					break;
-				default:
-					break;
-			}
-		} else if (typeof blockingPolicy === 'string') {
-			switch (blockingPolicy) {
-				case 'BLOCKING_POLICY_EVERYTHING':
-					decodedPolicy = 1;
-					break;
-				case 'BLOCKING_POLICY_RECOMMENDED':
-					decodedPolicy = 2;
-					break;
-				case 'BLOCKING_POLICY_NOTHING':
-					decodedPolicy = 3;
-					break;
-				default:
-					break;
-			}
+		switch (blockingPolicy) {
+			case BLOCKING_POLICY_EVERYTHING:
+				decodedPolicy = 1;
+				break;
+			case BLOCKING_POLICY_RECOMMENDED:
+				decodedPolicy = 2;
+				break;
+			case BLOCKING_POLICY_NOTHING:
+				decodedPolicy = 3;
+				break;
+			default:
+				break;
```

Comment:
If I'm not mistaken, we can dispense with the need for even this simplified converter function by setting the values of the `BLOCKING_POLICY_*` to be these numbers right in `share-hub\constants\BlockingPolicyConstants`, no?

Reply:
Since it looks like `background.js` is expecting a string, and another part of the app in the old hub is also sending it this way, we might have to make more changes than initially intended to implement this change. I think it might be best to leave it as is for now, and revisit it later in a wider refactor? I agree that it's worth looking into and keeping in mind.

Reply:
Sounds good